assets/js/modules/adsense/common/ad-blocker-warning.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * AdSense AdBlockerWarning component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { SvgIcon } from '../../../util';
+import { STORE_NAME } from '../datastore';
+const { useSelect } = Data;
+
+export default function AdBlockerWarning( { context } ) {
+	const isAdBlockerActive = useSelect( ( select ) => select( STORE_NAME ).isAdBlockerActive() );
+	const isAccountSetupComplete = useSelect( ( select ) => select( STORE_NAME ).getAccountSetupComplete() );
+	const isSiteSetupComplete = useSelect( ( select ) => select( STORE_NAME ).getSiteSetupComplete() );
+
+	// Return nothing if loading or if everything is fine.
+	if ( ! isAdBlockerActive ) {
+		return null;
+	}
+
+	let message;
+	if ( isAccountSetupComplete && isSiteSetupComplete ) {
+		message = __( 'Ad blocker detected, you need to disable it to get the AdSense latest data.', 'google-site-kit' );
+	} else {
+		message = __( 'Ad blocker detected, you need to disable it in order to setup AdSense.', 'google-site-kit' );
+	}
+
+	return (
+		<div
+			className={ classnames(
+				'googlesitekit-settings-module-warning',
+				{ [ `googlesitekit-settings-module-warning--${ context }` ]: context }
```

Comment:
What is `context` here?

Reply:
This is some historic thing, there are different CSS classes for the ad blocker warning rendered in different areas. The `context` itself is passed by legacy code that renders the warning in different places, see the `googlesitekit.ModuleSettingsWarning` filter.